### PR TITLE
add cls and TID

### PR DESCRIPTION
**Pull request by anandkumarpatel**
* add cls to add tid's to namespaces
* if job does not have tid, add it
* update examples to showoff 🌟  this new functionality 

- [ ] @bkendall

### CODE REVIEW

File: interfaces/modules/cls-bluebird.js.flow [hunk 1]
```diff
@@ -0,0 +1,5 @@
+declare module 'cls-bluebird' {
+  declare var exports: {
+    (cls: CLS): void;
+  }
+}
```
Comment (bkendall) [inline interfaces/modules/cls-bluebird.js.flow]: how do you not have newlines?!

File: interfaces/modules/continuation-local-storage.js.flow [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  static createNamespace(name: string): CLS;
+}
+
+declare module 'continuation-local-storage' {
+  declare var exports: Class<CLSClass>
+}
+
+export {
+  CLS
+}
```
Comment (bkendall) [inline interfaces/modules/continuation-local-storage.js.flow]: how do you not have newlines?!

File: interfaces/modules/uuid.js.flow [hunk 3]
```diff
@@ -0,0 +1,5 @@
+declare module 'uuid' {
+  declare var exports: {
+    (): string;
+  }
+}
```
Comment (bkendall) [inline interfaces/modules/uuid.js.flow]: how do you not have newlines?!

File: src/worker.js [hunk 4]
```diff
@@ -44,14 +46,20 @@ class Worker {
   log: any;
   msTimeout: any;
   queue: String;
+  tid: String;
+  attempt: number;
+  task: Function;
+  done: Function;
+  retryDelay: number;
```
Comment (bkendall) [inline src/worker.js]: can you re-sort these, please?

File: src/worker.js [hunk 5]
```diff
@@ -44,14 +46,20 @@ class Worker {
...
 
   constructor (opts: Object) {
     // managed required fields
     const fields = [
       'done',
       'job',
       'queue',
-      'task'
+      'task',
+      'log'
```
Comment (bkendall) [inline src/worker.js]: sort these as well

File: src/worker.js [hunk 6]
```diff
@@ -60,20 +68,22 @@ class Worker {
...
       // other options
       attempt: 0,
       msTimeout: process.env.WORKER_TIMEOUT || 0,
       retryDelay: process.env.WORKER_MIN_RETRY_DELAY || 1
     })
 
+    const tid = opts.job.tid || uuid()
+    opts.log = opts.log.child({ tid: tid, module: 'ponos:worker' })
     // put all opts on this
+    console.log('log', opts, this.log)
```
Comment (bkendall) [inline src/worker.js]: I don't think you want this log

**Comment by bkendall** [comment]
I would like you to put in the readme a block about how to use this. That would be wonderful. :)

**Comment by rsandor** [comment]
This seems fine

File: src/rabbitmq.js [hunk 7]
```diff
@@ -233,6 +214,36 @@ class RabbitMQ {
...
+    // flowtype does not prevent users from using this function incorrectly.
+    if (!isString(name) || name === '') {
+      throw new Error('name must be a string')
+    }
+    if (!isObject(content)) {
+      throw new Error('content must be an object')
+    }
+    // add tid to message if one does not exist
+    if (!content.tid) {
+      const ns = getNamespace('ponos')
```
Comment (bkendall) [inline src/rabbitmq.js]: `getNamespace` can return null? if so, change the interface to reflect that:

```
static getNamespace(name: string): ?CLS;
```

File: src/rabbitmq.js [hunk 8]
```diff
@@ -233,6 +214,36 @@ class RabbitMQ {
...
   /**
+   * Validate publish params. Also add tid to job if one does not exist
+   * @param  {Strong} name:    string        Name of queue or exchange
+   * @param  {Object} content: Object        content Content to send.
+   * @return {Buffer}          Content to send in job
+   * @throws {Error} If not connected
+   * @throws {Error} If name is not a string
+   * @throws {Error} If content is not an object
+   */
+  _validatePublish (name: string, content: Object) {
```
Comment (bkendall) [inline src/rabbitmq.js]: why do you need a name? are you just passing this because it's all the parameters for the publish methods? seems unnecessary...

Comment (anandkumarpatel) [inline src/rabbitmq.js]: I am validating name as well
```
+    if (!isString(name) || name === '') {
 +      throw new Error('name must be a string')
 +    }
```

Comment (bkendall) [inline src/rabbitmq.js]: Ah, it's _really_ because you removed it from the other methods.

**Pull request merged**